ga4gh/datamodel/__init__.py

```diff
@@ -187,3 +218,21 @@ def _scanDataFiles(self, dataDir, patterns):
             atexit.register(_cleanupHtslibsMess, indexDir)
         if numDataFiles == 0:
             raise exceptions.EmptyDirException(dataDir, patterns)
+
+    def getFileHandle(self, dataFile):
+        if dataFile in self.memo_table:
```

Comment:
We don't really have a good framework for using singleton objects, like this cache, throughout the code... now might be a good time to add some kind of IOC functionality?

Although, if usage of the cache is confined to just this module, that may not be necessary.  @jeromekelleher  your proposal seems fine to me, modulo thread-safety